### PR TITLE
Initial configuration for mac devicelab staging.

### PR DESCRIPTION
**Pull request by godofredoc**
This PR enables a mac devicelab test running on luci and using the
staging environment.

Bug:
  https://github.com/flutter/flutter/issues/70258

### CODE REVIEW

File: config/devicelab_staging_config.star [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+    # Defines console views.
+    console_view_name = "devicelab_staging"
+    luci.console_view(
+        name = console_view_name,
+        repo = repos.FLUTTER,
+        refs = [r"refs/heads/master"],
+    )
+
+    # Defines prod schedulers
```
Comment (keyonghan) [inline config/devicelab_staging_config.star:36]: Nit: update doc?

File: config/devicelab_staging_config.star [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        repo = repos.FLUTTER,
+        refs = [r"refs/heads/master"],
+    )
+
+    # Defines triggering policy
+    triggering_policy = scheduler.greedy_batching(
+        max_batch_size = 20,
+        max_concurrent_invocations = 2,
+    )
+    # Defines framework prod builders
```
Comment (keyonghan) [inline config/devicelab_staging_config.star:50]: Do we plan to support other repos, like engine as well?

**Review by keyonghan — APPROVED** [review]
LGTM with nit.

File: config/generated/flutter/luci/cr-buildbucket.cfg [hunk 3]
```diff
@@ -6146,6 +6146,43 @@ buckets {
...
+        name: "devicelab/devicelab_drone"
+        cipd_package: "flutter/recipe_bundles/flutter.googlesource.com/recipes"
+        cipd_version: "refs/heads/master"
+        properties_j: "$depot_tools/osx_sdk:{\"sdk_version\":\"11a420a\"}"
+        properties_j: "$fuchsia/goma:{\"server\":\"rbe-prod1.endpoints.fuchsia-infra-goma-prod.cloud.goog\"}"
+        properties_j: "$kitchen:{\"emulate_gce\":true}"
+        properties_j: "$recipe_engine/isolated:{\"server\":\"https://isolateserver.appspot.com\"}"
+        properties_j: "$recipe_engine/swarming:{\"server\":\"https://chromium-swarm.appspot.com\"}"
+        properties_j: "clobber:false"
+        properties_j: "dependencies:[{\"dependency\":\"android_sdk\"},{\"dependency\":\"open_jdk\"},{\"dependency\":\"xcode\"},{\"dependency\":\"gems\"},{\"dependency\":\"ios_signing\"},{\"dependency\":\"swift\",\"version\":\"mcKOF86ehMsd3HNodm9m11lOTQ04p58iHeQ5uK-d8C4C\"}]"
```
Comment (jmagman) [inline config/generated/flutter/luci/cr-buildbucket.cfg:6166]: Probably doesn't need android_sdk or open_jdk, but defer to you if there's some value to uniformity.

**Review by jmagman — APPROVED** [review]
I'm not sure about the luci-ness of it all, but other than the android_sdk comment, the logic looks fine to me.

File: config/generated/flutter/luci/cr-buildbucket.cfg (hunk 3 above)
Comment (godofredoc) [reply config/generated/flutter/luci/cr-buildbucket.cfg]: They are not needed, I removed them.

File: config/devicelab_staging_config.star (hunk 2 above)
Comment (godofredoc) [reply config/devicelab_staging_config.star:50]: Probably yes, but we don't know when.

File: config/devicelab_staging_config.star (hunk 1 above)
Comment (godofredoc) [reply config/devicelab_staging_config.star]: Done

**Pull request merged**